scripts/gh_scripts/weekly_status_report.mjs

```diff
@@ -0,0 +1,425 @@
+        owner: 'internetarchive',
+        repo: 'openlibrary',
+        labels: `Needs: Response`,
+        per_page: 100,
+        headers: {
+          'X-GitHub-Api-Version': '2022-11-28'
+        }
+      })
+    let isUpToDate = true
+
+    for (const lead of leads) {
+        const leadIssuesAwaitingComments = issuesAwaitingComments.filter((issue) => {
+            for (const label of issue.labels) {
+                if (label.name === lead.leadLabel) {
+                    return true
+                }
+            }
+            return false
+        })
+        const searchResultsUrl = `https://github.com/internetarchive/openlibrary/issues?q=is%3Aopen+is%3Aissue+label%3A%22Needs%3A+Response%22+label%3A${encodeURIComponent('"' + lead.leadLabel + '"')}`
+        if (leadIssuesAwaitingComments.length > 0) {
+            output.push(`  • <${searchResultsUrl}|${leadIssuesAwaitingComments.length} issue(s)> need response from ${lead.slackId}`)
+            isUpToDate = false
+        }
+    }
+
+    if (isUpToDate) {
+        output.push('  _No issues awaiting comments from leads_')
+    }
+
+    return output
+}
+
+
+/**
+ * Prepares a message containing links to unassigned issues and pull requests.
+ *
+ * @param {Array<Record>} pullRequests Non-draft pull request records
+ * @returns {Promise<Array<string>>} Messages with links to our unassigned PRs
+ */
+ async function prepareUnassignedItems(pullRequests) {
+    const output = ['*Needs: Lead/Assignees*']
+
+    const issuesNeedingLeads = await octokit.paginate('GET /repos/{owner}/{repo}/issues', {
+        owner: 'internetarchive',
+        repo: 'openlibrary',
+        labels: `Needs: Lead`,
+        per_page: 100,
+        headers: {
+          'X-GitHub-Api-Version': '2022-11-28'
+        }
+      })
+    output.push(`  • <https://github.com/internetarchive/openlibrary/issues?q=is%3Aissue+is%3Aopen+label%3A%22Needs%3A+Lead%22|${issuesNeedingLeads.length} issue(s)> need leads assigned by team`)
+
+    const unassignedPrs = pullRequests.filter((pull) => !pull.assignee)
+    const renovatebotPullCount = unassignedPrs.filter((pull) => pull.user.login === 'renovate[bot]').length
+    const unassignedCount = unassignedPrs.length - renovatebotPullCount
+    output.push(`  • <https://github.com/internetarchive/openlibrary/pulls?q=is%3Apr+is%3Aopen+no%3Aassignee+-is%3Adraft+-author:app/renovate|${unassignedCount} unassigned PRs> + <https://github.com/internetarchive/openlibrary/pulls/app%2Frenovate|${renovatebotPullCount} renovatebot>`)
+
+    return output
+}
+
+
+/**
+ * Finds all issues with the "Needs: Triage" label, prepares a message containing a
+ * summary for each given lead, and returns the prepared messages.
+ *
+ * If a lead has no issues that require triage, we do not publish their summary to
+ * Slack.
+ *
+ * @param {Array<Lead>}
+ * @returns {Promise<Array<string>>} Messages about untriaged issues
+ */
+async function prepareUntriagedIssues(leads) {
```

Comment:
These `async` `prepare*` functions can be improved by passing the issues as a parameter.  Then, they would no longer need to be `async`, and would only have concern themselves with preparing sections of the Slack message.

If a reviewer agrees, I'll go ahead and push that change.